.github/workflows/ci.yml

```diff
@@ -62,6 +62,28 @@ jobs:
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
+
+      # This task MUST run AFTER the checkout
+      # This task loads the ssh agent and adds the cadence deploy key into the agent
+      # This enables our go processes to pull private cadence dependencies
+      - name: Load cadence deploy key for private repo
```

Comment:
does a workflow such as https://github.com/webfactory/ssh-agent work? just to reduce complexity?

Reply:
This GHA would likely work, and I had looked at this. I was hesitant to pass secrets to this agent, and considering there were other commands being run, I figured that I would condense it to a single bash task rather than split them.